extensions.html

```diff
@@ -277,18 +277,19 @@ <h2>
             [=a promise resolved with=] {{GamepadHapticsResult/"preempted"}}.
             </li>
             <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
-            is not `undefined`, [=queue a global task=] on the [=relevant
-            global object=] of [=this=] using the [=gamepad task source=] to
-            run the following steps:
+            is not `undefined`:
               <ol>
-                <li>[=Resolve=]
-                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
-                with {{GamepadHapticsResult/"preempted"}}.
+                <li>Let |effectPromise| be
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}.
                 </li>
                 <li>Set
                 [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
                 `undefined`.
                 </li>
+                <li>[=Queue a global task=] on the [=relevant global object=]
+                of [=this=] using the [=gamepad task source=] to [=resolve=]
+                |effectPromise| with {{GamepadHapticsResult/"preempted"}}.
+                </li>
```

Comment:
This code path ends with promise being resolved, right? 

```suggestion
                </li>
                <li>Terminate this algorithm.</li>
```

Reply:
This path is for preempting the currently playing effect to prepare for the new effect. So there are two promises, the old effect promise and the new effect promise. We want to resolve the first promise (if it exists) and then create the new one. If we terminate here then we won't start the new effect.